In [1]:
import os 

os.chdir("..")

In [2]:
from dotenv import load_dotenv
load_dotenv()

True

In [3]:
"""
step 1:
pip install -r requirements.txt:
requests>=2.31.0
python-dotenv>=1.0.0
colorama>=0.4.6
pyyaml>=6.0

step 2:
orchestrate env add -n my-orc-env -u https://api.au-syd.watson-orchestrate.cloud.ibm.com/instances/5b80a5df-041c-4fa9-af19-67a6256731e7

step 3:
python rag_agent.py
"""
#!/usr/bin/env python3
"""
Standalone IBM watsonx Orchestrate client with auto‑refreshing token,
fast polling, and a simple ask(question) → answer API.
"""


'\nStandalone IBM watsonx Orchestrate client with auto‑refreshing token,\nfast polling, and a simple ask(question) → answer API.\n'

In [ ]:
from src.agents.wxorc_agent import OrchestrateClient



client = OrchestrateClient()
questions = [
    "Please give me the details of my latest orthopedist claim for member ID HF62415739",
    "That information seems outdated. My doctor just submitted more paperwork—what’s happening now?",
    "I submitted my claim a week ago but haven't received any update. What should I do?",
    "Why does my claim status still show as “Under Review” after 10 days?"
]

In [10]:
q = questions[0]
answer = client.ask(q)
print(f"Q: {q}")
print(f"A: {answer}\n")

Q: Please give me the details of my latest orthopedist claim for member ID HF62415739
A: Hello , I’ve accessed our insurance knowledge base using your member ID HF62415739. Based on the information available, I can see you have a recent orthopedist claim in our system.
Here’s what I found: You have a claim submitted on July 15th for $185 with Dr. Martinez for an orthopedist visit. The claim has been processed and approved. Insurance has paid $160 to your provider after applying your $25 specialist copay. Your explanation of benefits should be mailed to you within 3-5 business days of processing.
Is this the claim you were asking about?



#### sample response 
Q: Please give me the details of my latest orthopedist claim for member ID HF62415739
A: Hello , I’ve accessed our insurance knowledge base using your member ID HF62415739. Based on the information available, I can see you have a recent orthopedist claim in our system.
Here’s what I found: You have a claim submitted on July 15th for $185 with Dr. Martinez for an orthopedist visit. The claim has been processed and approved. Insurance has paid $160 to your provider after applying your $25 specialist copay. Your explanation of benefits should be mailed to you within 3-5 business days of processing.
Is this the claim you were asking about?


In [6]:
from langchain_core.tools import tool 

@tool
def ask_orchestrate(question: str) -> str:
    """
    Ask a question to the Orchestrate client and get an answer. 
    Useful when user asks a question that requires knowledge base access. 
    """
    client = OrchestrateClient()

    return client.ask(question)

In [8]:
ask_orchestrate.invoke("What is the status of my claim for member ID HF62415739?")

'Hello , I’ve accessed our insurance knowledge base using your member ID HF62415739. Based on the information available, I can see you have a recent orthopedist claim in our system.\n\nHere’s what I found: You have a claim submitted on July 15th for $185 with Dr. Martinez for an orthopedist visit. The claim has been processed and approved. Insurance has paid $160 to your provider after applying your $25 specialist copay. Your explanation of benefits should be mailed to you within 3-5 business days of processing.\n\nIs this the claim you were asking about?'